### DISCUSSION TITLE
References in additional includes cannot be found

### DISCUSSION THREAD

**Discussion by yakatz** [jgm/pandoc#10068] [Q&A] [answered]
I am not totally sure if this is Pandoc or an extension

I need to change the way appendix chapters are numbers to meet style rules. I copied the built-in tex template and added the following at the end:
```tex
$body$

\chapterstyle{myappendix}
\renewcommand{\chaptername}{Appendix}
\renewcommand{\thechapter}{\Alph{chapter}}
%\cftaddtitleline{toc}{chapter}{Appendices}{}
\appendix
\addtocontents{toc}{\protect\setcounter{tocdepth}{0}}

$for(include-after)$
$include-after$

$endfor$
\backmatter

\end{document}
```

In my main `md` file, I have this:

    ---
    ... other metadata ...
    include-after: |
      ``` {.include shift-heading-level-by=-1}
      back-pages/1-appendix_a.md
      back-pages/2-appendix_b.md
      back-pages/3-appendix_c.md
      ```
    ---
    
    ``` {.include}
    chapters/1-intro.md
    ...
    chapters/99-conclusion.md
    ```

The back pages are included and the table of contents is properly built, but I can't add references to the contents of the appendices (although I can add references to the chapter titles).

Example appendix contains
```md
# `myprogram` Code Listing {#sec:a_coderef}

This is the code listing for the program...
...

![Control Flow Graph for `myprogram`](../visio_export/CFG - Basic - labeled.png){#fig:simple_cfg_a width=95%}

```

```md
Works: A code listing for `myprogram` with cross-references to its Configuration Flow Graph is in [appendix @sec:a_coderef].

Doesn't work: In the Control Flow Graph in [figure @fig:simple_cfg_a], 
```

**Command**
`pandoc --template=template-mytemplate.tex --lua-filter=tools/include-files.lua --lua-filter=tools/alignment.lua --filter=pandoc-crossref --citeproc --lua-filter=tools/text-crossrefs.lua --lua-filter=tools/short-captions.lua --standalone --pdf-engine=xelatex -N --highlight-style=haddock -o mypaper.pdf mypaper.md`

`include-files.lua` is from here: https://github.com/pandoc-ext/include-files/blob/c86e823ad6e40c594d7b5c83166271703d2d49b7/include-files.lua

**Pandoc version?**
Windows 11  
pandoc.exe 3.2.1

**Comment by jgm** [comment, marked as answer]
`citeproc` only runs over the document's body, so it won't affect things you put in `include-after`.

Since pandoc can pass through raw latex content in a markdown file, you could just put this at the beginning of your first appendix file.
````
```{=latex}
\chapterstyle{myappendix}
\renewcommand{\chaptername}{Appendix}
\renewcommand{\thechapter}{\Alph{chapter}}
%\cftaddtitleline{toc}{chapter}{Appendices}{}
\appendix
\addtocontents{toc}{\protect\setcounter{tocdepth}{0}}
```
````
And then put the appendix files on the command line as arguments, after `mypaper.md`.

**Reply by yakatz** [reply]
My final main `md` file looks like this:

    ``` {.include}
    chapters/1-intro.md
    ...
    chapters/99-conclusion.md
    ```
    
    ``` {.include}
    back-pages/0-appendix_meta.md
    back-pages/1-appendix_a.md
    back-pages/2-appendix_b.md
    back-pages/3-appendix_c.md
    ```

To make the table of contents work, I also had to go back to my template tex and remove all the other plumbing I had put in for the `include-after`, i.e:

```patch
-$if(include-after)$
 \renewcommand\cftappendixname{\appendixname~}
-$endif$
```
